google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerSqlAstTranslator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019-2023 Google LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+package com.google.cloud.spanner.hibernate;
+
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.sql.ast.tree.Statement;
+import org.hibernate.sql.ast.tree.predicate.LikePredicate;
+import org.hibernate.sql.exec.spi.JdbcOperation;
+
+/** We need a translator for the LIKE operator, as Cloud Spanner does not support ESCAPE clauses. */
+public class SpannerSqlAstTranslator<T extends JdbcOperation>
```

Comment:
* From the documentation, this class looks very specific to `LIKE` operator. But `SpannerSqlAstTranslator` makes it sound too generic? Should we change this to `SpannerSqlLikeOperatorTranslator` ?
* Or are we saying that `LIKE` operator is the only translator that we need today. And future translators would be added in the same class by overrding new base methods in the future?

Reply:
It's a class that extends from `org.hibernate.dialect.SpannerSqlAstTranslator`, which is defined in the core Hibernate repository (it is part of the `SpannerDialect` that is defined there). That class again is a sub-class of `AbstractSqlAstTranslator`. The reason that the name is so generic, is that the class is intended to contain translators for all SQL operations for a given dialect. Almost all is defined in the generic base class, as most SQL databases use very similar dialects. The only method(s) that we need to override are the ones where we deviate from the standard.

So the reason that is only contains an override for `LIKE` is that this is the only type of expression where Cloud Spanner deviates enough from the standard that we actually need an override. But it could be more in the future.